docs/topics/compatibility-guides/compatibility-guide-21.md

```diff
@@ -0,0 +1,479 @@
+perspective
+(for example, from Java) is out of the scope of this document.
+
+## Language
+
+<!--
+### Title
+
+> **Issue**: [KT-NNNNN](https://youtrack.jetbrains.com/issue/KT-NNNNN)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**:
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: report a warning
+> - 2.2.0: raise the warning to an error
+-->
+
+### Change the typeOf() function behavior on Kotlin/Native
+
+> **Issue**: [KT-70754](https://youtrack.jetbrains.com/issue/KT-70754)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: The behavior of the `typeOf()` function on Kotlin/Native is aligned with Kotlin/JVM
+> to ensure consistency across platforms. 
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: align the `typeOf()` function behavior on Kotlin/Native
+
+### Prohibit exposing types through type parameters' bounds
+
+> **Issue**: [KT-69653](https://youtrack.jetbrains.com/issue/KT-69653)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Exposing types with lower visibility through type parameter bounds is now prohibited,
+> addressing inconsistencies in type visibility rules.
+> This change ensures that bounds on type parameters follow the same visibility rules as classes, preventing issues like IR validation errors in JVM.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: report a warning for exposing types via type parameter bounds with lower visibility
+> - 2.2.0: raise the warning to an error
+
+### Prohibit inheriting an abstract var property alongside a val property with the same name
+
+> **Issue**: [KT-58659](https://youtrack.jetbrains.com/issue/KT-58659)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: If a class inherits an abstract `var` property from an interface and a `val` property with the same name from a superclass,
+> it now triggers a compilation error. This resolves runtime crashes caused by missing setters in such cases.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: report a warning (or an error in progressive mode) when a class inherits an abstract `var` property from an interface and a `val` property with the same name from a superclass
+> - 2.2.0: raise the warning to an error
+
+### Report error when accessing uninitialized enum entries
+
+> **Issue**: [KT-68451](https://youtrack.jetbrains.com/issue/KT-68451)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The compiler now reports an error when uninitialized enum entries are accessed during enum class or entry initialization.
+> This aligns behavior with member property initialization rules, preventing runtime exceptions and ensuring consistent logic.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: report an error when accessing uninitialized enum entries
+
+### Changes in K2 smart cast propagation
```

Comment:
```suggestion
### Change K2 smart cast propagation
```

Reply:
I see previous headings are verb-oriented, the same for previous compatibility guides.

Reply:
There are a few examples where it deviates from being verb-oriented - I think it makes it less severe